flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/ScalingMetricEvaluator.java

```diff
@@ -66,12 +66,15 @@ public Map<JobVertexID, Map<ScalingMetric, EvaluatedScalingMetric>> evaluate(
         var scalingOutput = new HashMap<JobVertexID, Map<ScalingMetric, EvaluatedScalingMetric>>();
         var metricsHistory = collectedMetrics.getMetricHistory();
         var topology = collectedMetrics.getJobTopology();
+        var excludeVertexIdList = conf.get(AutoScalerOptions.EXCLUDE_VERTEX_IDS);
 
         for (var vertex : topology.getVerticesInTopologicalOrder()) {
-            scalingOutput.put(
-                    vertex,
-                    computeVertexScalingSummary(
-                            conf, scalingOutput, metricsHistory, topology, vertex));
+            if (!excludeVertexIdList.contains(vertex.toHexString())) {
+                scalingOutput.put(
```

Comment:
I think this will break the algorithm unless the vertex is a sink. The autoscaler relies on traversing the DAG to compute the desired processing rates. If we include a non-sink vertex, none of the downstream vertices will be able to be scaled.

Reply:
I think you should add this logic in the ScalingExecutor which will allow to ignore a vertex while still scaling all downstream vertices.

Reply:
make sense, doing changes

Reply:
as discussed,  we will follow up with an additional change to ensure that the ignored vertices do not affect their downstream vertices.